brainiak/reprsimil/brsa.py

```diff
@@ -30,24 +32,158 @@
+        and the length scale of the GP, but not the variance of the GP
+        (the purpose of this function is to estimate the variance).
+    n_y: int, number of observations
+    tau_range: float,
+        The reasonable range of tau, the standard deviation of the
+        Gaussian Process imposed on y=f(x). tau_range is parameter
+        of the inverse-Gamma prior.
+        The smaller it is, the more penalization is imposed
+        on large variation of y.
+    Returns:
+    --------
+    tau2: The MAP estimation of tau^2 based on the prior on tau
+        and y_invK_y.
+    log_ptau: log(p(tau)) of the returned tau^2 based on the
+        inverse-Gamma prior.
+    """
+    tau2 = (y_invK_y + 2 * tau_range**2) / (2 * 2 + 2 + n_y)
+    log_ptau = scipy.stats.invgamma.logpdf(
+        tau2, scale=tau_range**2, a=2)
+    return tau2, log_ptau
+
+
+def prior_GP_var_half_cauchy(y_invK_y, n_y, tau_range):
+    """ Imposing a half-Cauchy prior onto the standard deviation (tau)
+        of the Gaussian Process which is in turn a prior imposed over
+        a function y = f(x).
+        The scale parameter of the half-Cauchy prior is tau_range.
+        The function returns the MAP estimate of tau^2 and
+        log(p(tau|tau_range)) for the estimated value of tau^2,
+        where tau_range describes the reasonable range of tau
+        in the half-Cauchy prior.
+        An alternative form of prior is inverse-Gamma prior on tau^2.
+        Inverse-Gamma prior penalizes for both very small and very
+        large values of tau, while half-Cauchy prior only penalizes
+        for very large values of tau.
+        For more information on usage, see description in BRSA class:
+        `.BRSA`
+    """
+    tau2 = (y_invK_y - n_y * tau_range**2
+            + np.sqrt(n_y**2 * tau_range**4 + (2 * n_y + 8)
+                      * tau_range**2 * y_invK_y + y_invK_y**2))\
+        / 2 / (n_y + 2)
+    log_ptau = scipy.stats.halfcauchy.logpdf(
+        tau2**0.5, scale=tau_range)
+    return tau2, log_ptau
+
+
+def Ncomp_BIC_Minka(X):
+    """ This implements the BIC metric written in Minka 2000:
+        Automatic choice of dimensionality for PCA
+        The paper mainly proposed a Laplace approximation of the posterior
+        of the number of components but also provided this BIC approximation.
+        For cases in which feature number is smaller than sample size,
+        there is ambiguity for the author of this code to decide how to
+        set some variables. Empirically the number of components estimated
+        turn to be very large by the Laplace approximation method. The BIC
+        method appear to perform well in a limited number of datasets.
+    Parameters
+    ----------
+    X: 2-D numpy array of size [n_T, n_V]
+        The data to estimate the dimensionality for PCA.
+        Each column is one feature. Each row is one sample.
+        X is z-scored before further calculation.
```

Comment:
In general that would be fine -- but you have instructions in a few other docstrings to "zscore here the same way you zscore elsewhere" -- so I would say if you always zscore, just always zscore everywhere. I still think letting the user handle the z-score or passing as an option is better.